week3/assignment/src/components/ByType/DegreeSelect.jsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from "react";
+import DATA from "../../assets/DATA";
+import { Question, SectionWrapper } from "../../style/MainStyle";
+import { PrevNextBtnWrapper, PrevNextBtn } from "../../style/ButtonStyle";
+import styled from "styled-components";
+import SoloOrGroupSelect from "./SoloOrGroup";
+import MoodSelect from "./MoodSelect";
+
+const DegreeSelect = (props) => {
+  const { isStart, isStartHandler, selectedMood, setIsStart } = props;
+  const degreeList = [...new Set(DATA.map((elm) => elm.degree))];
+
+  const [selectedDegree, setSelectedDegree] = useState("");
+  const [isDegreeSelected, setIsDegreeSelected] = useState(false);
+  const [degreeNext, setdegreeNext] = useState(false);
+  const [currentStep, setCurrentStep] = useState(2);
+
+  const handleBtnChange = (e) => {
+    setSelectedDegree(e.target.value);
+    setIsDegreeSelected(true);
+  };
+
+  const handleNextBtn = () => {
+    setdegreeNext(true);
+    setCurrentStep(currentStep + 1);
+  };
+
+  const handlePrevBtn = () => {
+    setCurrentStep(currentStep - 1);
+    setIsStart(true);
+    setdegreeNext(false);
+  };
+
+  return (
+    <>
+      {!degreeNext && currentStep === 2 && (
+        <SectionWrapper>
+          <Question>분위기의 정도가 어땠으면 해?</Question>
+          <DegreeBtnWrapper>
+            {degreeList.map((degree) => (
+              <DegreeBtn
+                key={degree}
+                value={degree}
+                className={degree === selectedDegree ? "active" : ""}
```

Comment:
```suggestion
                className={degree === selectedDegree && "active" }
```
이건 어땡?

Reply:
아니면은! 
```suggestion
               active = {degree === selectedDegree}
```
이렇게 해서..? 근데 혹시 className은 여기 왜 쓰는 건가용? style 변경이면 지금 내가 쓴 코드로 바꿔도 될 것 같은뎅!

Reply:
마쟈 아니면 스타일드 컴포넌트의 props로도 가능할 듯!!
```
const DegreeBtn = styled.button`
...
    background-color: ${({ selectedDegree }) => (degree === selectedDegree ? "#00897b" : "")}
    color: ${({ selectedDegree }) => (degree === selectedDegree ? "#fff" : "")}
  }
`;
```
대충 이런식......? https://styled-components.com/docs/basics 이거나 https://react.vlpt.us/styling/03-styled-components.html 요고 확인해봐!